enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,608 @@
+    metadata:
+      name: cluster-proxy
+      namespace: cluster1
+    spec:
+      installNamespace: open-cluster-management-agent-addon
+    status:
+      configReference:
+        group: proxy.open-cluster-management.io
+        version: v1alpha1
+        resource: ManagedProxyConfiguration
+        name: cluster-proxy
+        lastObservedGeneration: 1
+    ```
+
+    If user's each add-on needs its own configuration, user set up the configuration name in the `ManagedClusterAddOn`
+
+    ```yaml
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ClusterManagementAddOn
+    metadata:
+      name: submariner
+    spec:
+      addOnConfiguration:
+        configGroupResource:
+          group: submarineraddon.open-cluster-management.io
+          resource: SubmarinerConfig
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: SubmarinerConfig
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      IPSecIKEPort: 500
+      IPSecNATTPort: 4500
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      config:
+        name: submariner
+        namespace: cluster1
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: SubmarinerConfig
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      IPSecIKEPort: 501
+      IPSecNATTPort: 4501
+    
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      config:
+        name: submariner
+        namespace: cluster2
+    ```
+
+    then the configuration reference in this add-on status will be
+
+    ```yaml
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster1
+    spec:
+      config:
+        name: submariner
+        namespace: cluster1
+    status:
+      configReference:
+        group: proxy.open-cluster-management.io
+        version: v1alpha1
+        resource: SubmarinerConfig
+        name: submariner
+        namesapce: cluster1
+        lastObservedGeneration: 1
+
+    apiVersion: addon.open-cluster-management.io/v1alpha1
+    kind: ManagedClusterAddOn
+    metadata:
+      name: submariner
+      namespace: cluster2
+    spec:
+      config:
+        name: submariner
+        namespace: cluster2
+    status:
+      configReference:
+        group: proxy.open-cluster-management.io
+        version: v1alpha1
+        resource: SubmarinerConfig
+        name: submariner
+        namesapce: cluster2
+        lastObservedGeneration: 1
+    ```
+
+### Test Plan
+- All implementations will have thorough coverage of unit tests.
+- Integration tests will cover the user cases
+
+### Graduation Criteria
+1. The new APIs is reviewed and accepted;
```

Comment:
let's define alpha/beta stage here. I think we need some criteria for beta.
e.g.
1. at least 2-3 consumers of this API
2. at lease 2-3 customized API using this mechanism.

Reply:
added